mne/time_frequency/csd.py

```diff
@@ -1,63 +1,514 @@
+            used as the lower bounds (inclusive) of frequency bins and the mean
+            is taken for each bin.
+        fmax : float | list of float | None
+            Upper bound of the frequency range in Hertz. Defaults to the
+            highest frequency available. When a list of frequencies is given,
+            these are used as the upper bounds (inclusive) of frequency bins
+            and the mean is taken for each bin.
+
+        Returns
+        -------
+        csd : Instance of CrossSpectralDensity
+            The CSD matrix, averaged across the given frequency range(s).
+        """
+        csd = self.sum(fmin, fmax)
+        for i, f in enumerate(csd.frequencies):
+            csd._data[:, i] /= len(f)
+        return csd
+
+    def _get_frequency_index(self, freq):
+        """Find the index of the given frequency in ``self.frequencies``.
+
+        If the exact given frequency is not available, the nearest frequencies
+        will be chosen, up to a difference of 1 Hertz.
+
+        Parameters
+        ----------
+        freq : float
+            The frequency to find the index for
+
+        Returns
+        -------
+        index : int
+            The index of the frequency nearest to the requested frequency.
+        """
+        if self.is_sum:
+            raise ValueError('This CSD object represents a mean across '
+                             'frequencies. Cannot select a specific '
+                             'frequency.')
+
+        distance = np.abs(np.asarray(self.frequencies) - freq)
+        index = np.argmin(distance)
+        min_dist = distance[index]
+        if min_dist > 1:
+            raise IndexError('Frequency %f is not available.' % freq)
+        return index
+
+    def pick_frequency(self, freq=None, index=None):
+        """Get a CrossSpectralDensity object with only the given frequency.
+
+        Parameters
+        ----------
+        freq : float | None
+            Return the CSD matrix for a specific frequency. Only available
+            when no averaging across frequencies has been done.
+        index : int | None
+            Return the CSD matrix for the frequency or frequency-bin with the
+            given index.
+
+        Returns
+        -------
+        csd : instance of CrossSpectralDensity
+            A CSD object containing a single CSD matrix that corresponds to the
+            requested frequency or frequency-bin.
+
+        See Also
+        --------
+        get_matrix
+        """
+        if freq is None and index is None:
+            raise ValueError('Use either the "freq" or "index" parameter to '
+                             'select the desired frequency.')
+
+        elif freq is not None:
+            if index is not None:
+                raise ValueError('Cannot specify both a frequency and index.')
+
+            index = self._get_frequency_index(freq)
+
+        return self[index]
+
+    def get_matrix(self, frequency=None, index=None):
```

Comment:
maybe `get_array` so it's not confused with returning `np.matrix` type

Reply:
I meant it in the spirit of 'get the CSD matrix at frequency f', not as 'get a version of this CSD object in the form of a numpy array'. I could rename it to `get_csd_mat`: would that work for you?